app/src/main/java/com/example/rholbrook/tickettoride/game/GameActivityPresenter.java

```diff
@@ -228,38 +226,46 @@ public void readyToInitialize() {
         TICKET_IMAGE_MAP.put(Integer.valueOf(30), R.mipmap.winnipeg_little_rock);
     }
 
-    public void runDemo1() {
-        mModel.runDemo1();
+    public void demo(int timesClicked) {
+        mModel.demo(timesClicked);
     }
 
-    public void runDemo2() {mModel.runDemo2();}
-
     @Override
     public void setDeckCount(int ticketDeckCount, int trainDeckCount) {
         viewCallback.updateDeckCounts(ticketDeckCount, trainDeckCount);
     }
 
     @Override
     public void setOpponentOneTurn(Player opponentOne) {
+        viewCallback.setClientNotTurn(mModel.getClient());
         viewCallback.setOpponentOneTurn(opponentOne);
     }
 
     @Override
     public void setOpponentTwoTurn(Player opponentTwo) {
+        viewCallback.setOpponentOneNotTurn(mModel.getClient());
         viewCallback.setOpponentTwoTurn(opponentTwo);
 
     }
 
     @Override
     public void setOpponentThreeTurn(Player opponentThree) {
+        viewCallback.setOpponentTwoNotTurn(mModel.getClient());
         viewCallback.setOpponentThreeTurn(opponentThree);
-
     }
 
     @Override
     public void setOpponentFourTurn(Player opponentFour) {
+        viewCallback.setOpponentThreeNotTurn(mModel.getClient());
         viewCallback.setOpponentFourTurn(opponentFour);
+    }
+
+    public void message(String message) {
+        viewCallback.message(message);
+    }
 
+    public void updatePlayerOne(Player player) {
+        viewCallback.updatePlayerOne(player);
     }
 
```

Comment:
If we are implementing the methods as you said, we don't have to allow the demo, right? Just making sure.

Reply:
If the server is notifying us when an opponent draws a card, takes tickets and selects routes, then yes, we do not need the demo